nodeletctl/pkg/nodeletctl/config.go

```diff
@@ -0,0 +1,127 @@
+package nodeletctl
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"go.uber.org/zap"
+)
+
+type NodeletConfig struct {
+	AllowWorkloadsOnMaster bool
+	CalicoV4Interface      string
+	CalicoV6Interface      string
+	ClusterId              string
+	ContainerRuntime       ContainerRuntimeConfig
+	EtcdClusterState       string
+	HostId                 string
+	HostIp                 string
+	K8sApiPort             string
+	MasterList             *map[string]string
+	MasterIp               string
+	MasterVipEnabled       bool
+	MasterVipInterface     string
+	MasterVipVrouterId     int
+	Mtu                    string
+	Privileged             string
+	NodeletRole            string
+	UserImages             []string
+	CoreDNSHostsFile       string
+	IPv6Enabled            bool
+	UseHostname            bool
+	CalicoIP4              string
+	CalicoIP6              string
+	CalicoV4BlockSize      int
+	CalicoV6BlockSize      int
+	CalicoV6ContainersCidr string
+	CalicoV4ContainersCidr string
+	CalicoV4NATOutgoing    bool
+	CalicoV6NATOutgoing    bool
+	CalicoV4IpIpMode       string
+	ContainersCidr         string
+	ServicesCidr           string
+}
+
+func setNodeletClusterCfg(cfg *BootstrapConfig, nodelet *NodeletConfig) {
```

Comment:
Why not just return the config object instead of pointer? It makes the function much more readable from the definition.